### PR TITLE
New responses

### PR DESCRIPTION
**Pull request by JJerome-NM**
1. Add a message sending controller
2. Change mappings from "request - response" to just "request".
3. Add filter order.
4. Change the response header.

### CODE REVIEW

File: jjsocket-main/src/main/java/com/jjerome/context/SocketControllersContext.java [hunk 1]
```diff
@@ -1,13 +1,18 @@
 package com.jjerome.context;
 
-import com.jjerome.annotations.SocketConnectMapping;
-import com.jjerome.annotations.SocketDisconnectMapping;
-import com.jjerome.annotations.SocketMapping;
+import com.jjerome.annotations.*;
```
Comment (justme1135) [inline jjsocket-main/src/main/java/com/jjerome/context/SocketControllersContext.java:3]: Remove single imports

File: jjsocket-main/src/main/java/com/jjerome/context/SocketControllersContext.java [hunk 2]
```diff
@@ -19,10 +24,13 @@
 import java.util.*;
```
Comment (justme1135) [inline jjsocket-main/src/main/java/com/jjerome/context/SocketControllersContext.java:24]: Same here

File: jjsocket-main/src/main/java/com/jjerome/filters/FiltersComparator.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+public class FiltersComparator<F> implements Comparator<F> {
+    @Override
+    public int compare(@NotNull F f1, @NotNull F f2) {
+        if (!f1.getClass().isAnnotationPresent(FilteringOrder.class)) return 1;
+        if (!f2.getClass().isAnnotationPresent(FilteringOrder.class)) return -1;
+
+        int result = f1.getClass().getDeclaredAnnotation(FilteringOrder.class).order() -
+                f2.getClass().getDeclaredAnnotation(FilteringOrder.class).order();
+
+        return result != 0 ? result : f1 == f2 ? 0 : -1;
```
Comment (justme1135) [inline jjsocket-main/src/main/java/com/jjerome/filters/FiltersComparator.java:17]: Rewrite it so it becomes more readable and not so complex

File: jjsocket-main/src/main/java/com/jjerome/models/SocketApplication.java [hunk 4]
```diff
@@ -19,92 +26,139 @@
...
+
+    private final ExecutorService executorService = Executors.newFixedThreadPool(THREAD_POOL);
 
     private final Map<String, WebSocketSession> allSession = new HashMap<>();
 
-    private final Map<String, Function<TextMessage, Response<?>>> socketMappings = new HashMap<>();
+//    This thing has to be here, otherwise everything will break
+    {
+        messageSender = new MessageSender(this.allSession, this.executorService);
+    }
```
Comment (justme1135) [inline jjsocket-main/src/main/java/com/jjerome/models/SocketApplication.java:46]: Refactor without using initialization blocks

**Review by justme1135 — CHANGES_REQUESTED** [review]

File: jjsocket-main/src/main/java/com/jjerome/context/SocketControllersContext.java (hunk 1 above)
Comment (JJerome-NM) [reply jjsocket-main/src/main/java/com/jjerome/context/SocketControllersContext.java]: I modified it for normal imports

File: jjsocket-main/src/main/java/com/jjerome/context/SocketControllersContext.java (hunk 2 above)
Comment (JJerome-NM) [reply jjsocket-main/src/main/java/com/jjerome/context/SocketControllersContext.java]: This is also fixed here

File: jjsocket-main/src/main/java/com/jjerome/mappers/ResponseMapper.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        if (validateJsonField(jsonObject, "requestBody", reqBodyClass)){
+            if (jsonFieldIsNumberOrString(jsonObject, "requestBody")){
+                response.setResponseBody((RB) jsonObject.opt("requestBody"));
+                return response;
+            }
+            response.setResponseBody(JsonMapper.map(reqBodyClass, jsonObject.getJSONObject("requestBody")));
+            return response;
+        }
+        response.setResponseBody(null);
+        return response;
```
[the remark this answers is not in the document]
Comment (JJerome-NM) [reply jjsocket-main/src/main/java/com/jjerome/mappers/ResponseMapper.java:31]: Made fewer retours

File: jjsocket-main/src/main/java/com/jjerome/filters/FiltersComparator.java (hunk 3 above)
Comment (JJerome-NM) [reply jjsocket-main/src/main/java/com/jjerome/filters/FiltersComparator.java:19]: I made it a little simpler

File: jjsocket-main/src/main/java/com/jjerome/models/SocketApplication.java (hunk 4 above)
Comment (JJerome-NM) [reply jjsocket-main/src/main/java/com/jjerome/models/SocketApplication.java]: Removed this initialization block

File: jjsocket-main/src/main/java/com/jjerome/filters/FiltersComparator.java (hunk 3 above)
Comment (justme1135) [reply jjsocket-main/src/main/java/com/jjerome/filters/FiltersComparator.java:19]: If you do the `f1 == f2` check and return directly after line 12 it becomes easier to read

File: jjsocket-main/src/main/java/com/jjerome/annotations/FilteringOrder.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package com.jjerome.annotations;
+
+import java.lang.annotation.*;
```
Comment (justme1135) [inline jjsocket-main/src/main/java/com/jjerome/annotations/FilteringOrder.java:3]: Remove wildcard imports

File: jjsocket-main/src/main/java/com/jjerome/annotations/SocketComponentsScan.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package com.jjerome.annotations;
+
+import java.lang.annotation.*;
```
Comment (justme1135) [inline jjsocket-main/src/main/java/com/jjerome/annotations/SocketComponentsScan.java:3]: Same here

File: jjsocket-main/src/main/java/com/jjerome/annotations/SocketMappingFilters.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
+package com.jjerome.annotations;
+
+import com.jjerome.filters.SocketMethodFilter;
+
+import java.lang.annotation.*;
```
Comment (justme1135) [inline jjsocket-main/src/main/java/com/jjerome/annotations/SocketMappingFilters.java:5]: Same here

**Review by justme1135 — CHANGES_REQUESTED** [review]

File: jjsocket-main/src/main/java/com/jjerome/annotations/FilteringOrder.java (hunk 6 above)
Comment (JJerome-NM) [reply jjsocket-main/src/main/java/com/jjerome/annotations/FilteringOrder.java]: Removed wildcard imports

File: jjsocket-main/src/main/java/com/jjerome/annotations/SocketComponentsScan.java (hunk 7 above)
Comment (JJerome-NM) [reply jjsocket-main/src/main/java/com/jjerome/annotations/SocketComponentsScan.java]: Removed wildcard imports

File: jjsocket-main/src/main/java/com/jjerome/annotations/SocketMappingFilters.java (hunk 8 above)
Comment (JJerome-NM) [reply jjsocket-main/src/main/java/com/jjerome/annotations/SocketMappingFilters.java]: Removed wildcard imports

File: jjsocket-main/src/main/java/com/jjerome/filters/FiltersComparator.java (hunk 3 above)
Comment (JJerome-NM) [reply jjsocket-main/src/main/java/com/jjerome/filters/FiltersComparator.java]: Added a check after line 12

File: jjsocket-main/src/main/java/com/jjerome/models/RequestAccepter.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+    private final MessageSender MESSAGE_SENDER = SocketApplication.getMessageSender();
+
+    private final Map<String, BiConsumer<WebSocketSession ,TextMessage>> methodMappings;
+
+    private final Set<SocketMessageFilter> messageFilters;
+
+    private final ExecutorService executorService;
+
+    public RequestAccepter(Map<String, BiConsumer<WebSocketSession ,TextMessage>> methodMappings,
```
Comment (justme1135) [inline jjsocket-main/src/main/java/com/jjerome/models/RequestAccepter.java:28]: Replace with `@RequiredArgsConstructor` on class

File: jjsocket-main/src/main/java/com/jjerome/models/ResponseErrors.java [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+    MAPPING_ERROR("/error/socket/mapping", "Socket mapping error"),
+    ;
+
+    private final Response<String> response;
+
+    ResponseErrors(String path, String message) {
+        this.response = new Response<>(path, message);
+    }
+
+    public Response<String> get(){
```
Comment (justme1135) [inline jjsocket-main/src/main/java/com/jjerome/models/ResponseErrors.java:20]: Replace explicit getter with `@Getter` annotation

File: .gitignore [hunk 11]
```diff
@@ -3,6 +3,7 @@ target/
 !.mvn/wrapper/maven-wrapper.jar
 !**/src/main/**/target/
 !**/src/test/**/target/
+MainTests
```
Comment (justme1135) [inline .gitignore:6]: This entry bothers me a little. So you have some tests set up and you explicitly ignore them? Would not it be better just to remove them?

**Review by justme1135 — CHANGES_REQUESTED** [review]
Few small fixes and you are good to go

File: jjsocket-main/src/main/java/com/jjerome/models/RequestAccepter.java [hunk 12]
```diff
@@ -0,0 +1,57 @@
+package com.jjerome.models;
+
+import com.jjerome.filters.SocketMessageFilter;
+import org.jetbrains.annotations.NotNull;
```
[the remark this answers is not in the document]
Comment (JJerome-NM) [reply jjsocket-main/src/main/java/com/jjerome/models/RequestAccepter.java]: I fixed it, unfortunately I didn't find it before

File: jjsocket-main/src/main/java/com/jjerome/models/RequestAccepter.java (hunk 9 above)
Comment (JJerome-NM) [reply jjsocket-main/src/main/java/com/jjerome/models/RequestAccepter.java]: Added this annotation, very good recommendation

File: jjsocket-main/src/main/java/com/jjerome/models/ResponseErrors.java (hunk 10 above)
Comment (JJerome-NM) [reply jjsocket-main/src/main/java/com/jjerome/models/ResponseErrors.java]: Removed that method and added the @Getter annotation

File: .gitignore (hunk 11 above)
Comment (JJerome-NM) [reply .gitignore]: They were bad tests, so I didn't want them to hit the gate. Although I think I removed them in the previous commit

**Review by justme1135 — APPROVED** [review]

**Pull request merged**